torch/distributed/_functional_collectives.py

```diff
@@ -4,13 +4,18 @@
 import torch.distributed as dist
 import torch.distributed.distributed_c10d as c10d
 from typing import Tuple, Union, List, cast, TYPE_CHECKING
-from torch.utils._pytree import tree_map_only
 from . import _functional_collectives_impl as fun_col_impl
 from ._functional_collectives_impl import _register_tensor_wrapper
 from torch.fx.experimental.proxy_tensor import (
     get_innermost_proxy_mode,
 )
 
+try:
+    from torch.utils._cxx_pytree import tree_map_only
+except ImportError:
+    tree_map_only = torch.utils._pytree.tree_map_only  # type: ignore[assignment]
```

Comment:
do we know under what builds/setups this import won't exist? Should we issue a warning if we're in a known slow mode?

Reply:
I believe some certain CI does not have access to `pip` channel and it only have conda channel, where optree not yet available in conda. @XuehaiPan might have more details

Reply:
I'm making `optree` optional in PR #109684. We don't need this try-except blocks after PR #109684.

- #109684